src/event_system/events.ts

```diff
@@ -0,0 +1,105 @@
+import Queue = require('better-queue')
+import { StoreOptions } from "better-queue"
+import { ConnectionStringParser } from "connection-string-parser";
+import { EventT, EventsT } from "../types/events/eventT";
+import { RegisteredEventT, RegisteredEventsT } from "../types/events/registered_eventT";
+import { TaskT, TasksT } from "../types/events/taskT";
+import EventStorage from "../data_storage/events/event_storage";
+import QueueStorage from "../data_storage/events/queue_storage";
+import Result from "../result";
+import Logger from "../logger";
+import axios, { AxiosResponse } from "axios";
+import Config from "../config";
+
+export class QueueProcessor {
+
+  private static instance: QueueProcessor;
+
+  public static get Instance(): QueueProcessor {
+    if (!QueueProcessor.instance) {
+      QueueProcessor.instance = new QueueProcessor()
+    }
+
+    return QueueProcessor.instance
+  }
+
+  private store = makeStore();
+
+  // define the database queue
+  private messageQueue: Queue = new Queue(() => undefined, {
+    store: this.store
+  });
+
+  public addEvents(events: EventsT) {
```

Comment:
have we thought of a single word name for this function? Since it's part of the class I don't think we need to repeat `Events` in the name. I'm fine with either `add` or `emit`